include/libdnf5-cli/output/advisorylist.hpp

```diff
@@ -34,9 +34,20 @@ LIBDNF_CLI_API void print_advisorylist_table(
     std::vector<std::unique_ptr<IAdvisoryPackage>> & advisory_package_list_not_installed,
     std::vector<std::unique_ptr<IAdvisoryPackage>> & advisory_package_list_installed);
 
+LIBDNF_CLI_API void print_advisorylist_json(
+    std::vector<std::unique_ptr<IAdvisoryPackage>> & advisory_package_list_not_installed,
+    std::vector<std::unique_ptr<IAdvisoryPackage>> & advisory_package_list_installed);
+
 LIBDNF_CLI_API void print_advisorylist_references_table(
     std::vector<libdnf5::advisory::AdvisoryPackage> & advisory_package_list_not_installed,
     std::vector<libdnf5::advisory::AdvisoryPackage> & advisory_package_list_installed,
+    std::string pretty_reference_type,
```

Comment:
okay, I'll keep the `reference_type` and move the pretty name over to a separate function it can call.

Since `updateinfo.xml` in the wild lists a lot more than `bugzilla` and `cve`, see https://github.com/rpm-software-management/dnf5/blob/65076dc83ecccf3d75d18271c8922a7ab3c0cef1/doc/repository_format/updateinfo-permissive.xsd#L506-L531 

Thus, for a future patch, I want to support these too.... just not sure how we would want to deal with the pretty names for them.

Reply:
I was thinking about it and personally I would just drop the pretty name from the json output. For the plain text output it is just a column header to let the user know what it means but in json output the type information is already there. Plus we have to guess how it should look like.

I like the fallback to the type string (even for the plain text). :+1: 

Otherwise this looks quite good to me. I have [made an issue](https://github.com/rpm-software-management/ci-dnf-stack/issues/1513) to add tests to our CI and I think we can merge this.

Reply:
The CI is a somewhat complicated. You can read the [README](https://github.com/rpm-software-management/ci-dnf-stack/blob/main/README.md) for general overview and how to run it in a container.

Its also possible to run directly (which is fine in this case since these commands don't modify the host system in any way) described [here](https://github.com/rpm-software-management/ci-dnf-stack/tree/main/dnf-behave-tests). Though its a bit outdated, for dnf5 the behave command should look something like:
```
behave --tags @dnf5 -Ddnf_command=dnf5 ./ci-dnf-stack/dnf-behave-tests/dnf/updateinfo.feature
```
Dependencies from `requirements.spec` and python `requirements.txt` are needed.

For the actual tests I would say the features are quite self explanatory, look at: https://github.com/rpm-software-management/ci-dnf-stack/blob/main/dnf-behave-tests/dnf/updateinfo.feature

Reply:
I should add that the build with the `advisory list --json` option will be available in https://copr.fedorainfracloud.org/coprs/rpmsoftwaremanagement/dnf-nightly/builds/ tomorrow so if you build the container today by default it won't be there.